ui/pages/confirmations/components/confirm/info/shared/transaction-details.tsx

```diff
@@ -0,0 +1,78 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { isValidAddress } from 'ethereumjs-util';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowText,
+  ConfirmInfoRowUrl,
+} from '../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+import { useKnownMethodDataInTransaction } from '../hooks/known-method-data-in-transaction';
+
+const OriginRow = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  return currentConfirmation?.origin ? (
+    <ConfirmInfoRow
+      label={t('requestFrom')}
+      tooltip={t('requestFromTransactionDescription')}
+    >
+      <ConfirmInfoRowUrl url={currentConfirmation.origin} />
```

Comment:
Minor, but should we use a variable like `origin` for repeated property chains like this?